data-model/src/main/java/io/micronaut/data/model/Page.java

```diff
@@ -63,6 +63,32 @@ default int getTotalPages() {
         return size == 0 ? 1 : (int) Math.ceil((double) getTotalSize() / (double) size);
     }
 
+    /**
+     * Determines whether there is a next page.
+     *
+     * @since 4.8.0
+     * @return Whether there exist a next page.
+     */
+    default boolean hasNext() {
+        if (getPageable() instanceof CursoredPageable cursoredPageable) {
+            return cursoredPageable.hasNext();
+        }
+        return getOffset() + getSize() < getTotalSize();
```

Comment:
Can we also have hasNext/hasPrevious in `Pageable` and just delegate it?

Reply:
Pageable does not know the total size. Do you want to delegate `hasNext(getTotalSize())`? That would be strange since the method would have to be in public API.